lib/logstash/inputs/snmptrap.rb

```diff
@@ -47,26 +47,44 @@ def register
     end
   end # def register
 
-  public
   def run(output_queue)
     begin
       # snmp trap server
       snmptrap_listener(output_queue)
     rescue => e
       @logger.warn("SNMP Trap listener died", :exception => e, :backtrace => e.backtrace)
-      sleep(5)
-      retry
+      stoppable_call(20) { sleep(0.25) }
```

Comment:
Yep, been there and I was bite by it too, See:
https://github.com/jordansissel/ruby-stud/pull/27#issuecomment-140590272

I think we need a common way of doing it or the pipeline should provide it or enforce a strategies.